### PR TITLE
Fine-tune MobileBert on smaller batches in its test

### PR DESCRIPTION
**Pull request by shoumikhin**
## Summary

The MobileBert test in the Samsung model suite is killed for using too much memory.
That stops the whole suite, and because the suite is a required check, it stops the
branch nightly builds are cut from moving forward.

The test asks for a fine-tuned model, and fine-tuning means real training: five passes
over a sentiment dataset, sixty four sentences at a time. Training keeps every
intermediate value until the backward pass finishes, and a sentence of a hundred and
twenty eight tokens through twenty four layers produces a lot of them.

Measured on a single training step, same model and sequence length:

```
batch 64   peak memory 7.58 GiB
batch  8   peak memory 1.43 GiB
```

Smaller batches take longer but hold far less at once, and they do not change what the
model learns, only how often it steps. So the batch size becomes an argument, today's
value stays the default, and the test asks for a smaller one.

## What this deliberately does not change

The number of passes is left alone. Cutting it would shorten the test further, but the
output range the model ends up with is not a falling function of it. Measured on a fixed
probe, the largest logit is about two million untuned, about half a million after one
pass, and about two and a half million after two. Fewer passes is therefore not
obviously safer for the backend, so the memory problem is solved without touching it.

## Test plan

Measured the peak memory of one training step at both batch sizes, on Linux, with the
same model and sequence length: 7.58 GiB against 1.43 GiB, a bit over five times less.

Ran the suite's own order in one process to find where the peak comes from. Two vision
models ahead of this one reach half a gigabyte between them, so the training step is the
peak rather than an accumulation across models.

Confirmed the example script is unchanged. It passes only the artifacts directory, so
the batch size falls back to its default and the accuracy it prints is produced exactly
as before.

### CODE REVIEW

**Review by sylvesterkaczmarek — COMMENTED** [review]
Making batch size and epoch count explicit parameters is a reasonable way to keep the example defaults unchanged while bounding the milestone test's memory cost. The remaining risk is that the test now relies on one epoch being sufficient to bring activation ranges into the backend's representable regime. If that property is load-bearing, I would assert the relevant range/compile condition directly so a data or optimiser change fails diagnostically rather than as a later backend mismatch.

**Comment by shoumikhin** [comment]
You are right, and I have dropped that half of the change.

I measured whether the number of passes controls the output range, on a fixed probe,
and it does not. The largest logit is about two million untuned, about half a million
after one pass, and about two and a half million after two. Not a falling function, so
cutting the passes would have changed a property I cannot argue for.

The change is now only the batch size. That is what the memory measurement supports:
seven and a half gigabytes at sixty four sentences against one and a half at eight, on
one training step. Smaller batches do not change what the model learns, only how often
it steps, so nothing about the range depends on this.

On asserting the range directly, I agree that would be better than leaving it implicit,
and I have not added it here. My numbers above are from random tokens rather than the
real dataset, so I do not yet know what a fair threshold would be, and a wrong one would
fail this test for reasons that have nothing to do with the backend. That belongs with
someone who knows what range the backend can actually represent, and it is worth doing
on its own rather than inside a memory fix.

**Review by JacobSzwejbka — APPROVED** [review]
